tests/create_test_dataset.py

```diff
@@ -0,0 +1,145 @@
+
+# These settings will indicate what data you want to read in from the AWS catalog. 
+# The variables will be read in individually using the _read_from_catalog function, which returns an xarray DataArray. At the end, we will combine the individual DataArrays to form a single xarray Dataset object.
+
+# ---- Settings to generate testing file test_dataset_2022_2022_monthly_45km 
+variable_list = ["RAINC", "RAINNC", "SWDDIF", "TSK", "PSFC", "T2", "Q2", "U10", "V10", "SNOWNC", "LWDNB", "LWDNBC", "LWUPB", "LWUPBC", "SWDNB", "SWDNBC", "SWUPB", "SWUPBC"] 
+year_start = 2022 # Starting year for time slice (integer)
+year_end = 2022 # Ending year for time slice (integer)
+timescale = "monthly" # Timescale (string): hourly, daily, or monthly
+resolution = "45 km" # Resolution (string): 3 km, 9 km , or 45 km 
+append_historical = False # Append historical data? (boolean) year_start must be < 2015
+scenarios = ['SSP 2-4.5 -- Middle of the Road', 'SSP 3-7.0 -- Business as Usual', 'SSP 5-8.5 -- Burn it All']
+
+
+# ---- Settings to generate testing file timeseries_data_T2_2014_2016_monthly_45km.nc 
+#variable_list = ["T2"] 
+#year_start = 2014 
+#year_end = 2016 
+#timescale = "monthly" 
+#resolution = "45 km" 
+#append_historical = True 
+#scenarios = ['SSP 2-4.5 -- Middle of the Road', 'Historical Climate']
+
+
+# -----------------  READ IN DATA FROM AWS CATALOG -----------------
+
+def read_data_for_var(variable, year_start=2013, year_end=2016, append_historical=True, timescale="monthly", resolution="45 km", scenarios=['SSP 2-4.5 -- Middle of the Road', 'SSP 3-7.0 -- Business as Usual', 'SSP 5-8.5 -- Burn it All']): 
+    """ Read data from catalog for a given variable. """
+    selections = DataSelector(append_historical=append_historical, 
+                              area_average=False, 
+                              name='DataSelector00101', 
+                              resolution=resolution, 
+                              scenario=scenarios, 
+                              time_slice=(year_start, year_end), 
+                              timescale=timescale, 
+                              variable=variable)
+    location = LocSelectorArea(area_subset='none', 
+                               cached_area='CA', 
+                               latitude=(32.5, 42), 
+                               longitude=(-125.5, -114), 
+                               name='LocSelectorArea00102')
+    xr_da = _read_from_catalog(selections=selections, location=location)
+    return xr_da
+
+def progressBar(i, tot): 
+    """Display a progress bar inside a for loop 
+    Based on Stack Overflow answer from Mark Rushakoff: https://stackoverflow.com/questions/3002085/how-to-print-out-status-bar-and-percentage
+    
+    Args: 
+        i (int): iteration number
+        tot (int): total number of iterations
+    """
+    j = (i+1) / tot
+    sys.stdout.write('\r[%-20s] %d%% complete' % ('='*int(20*j), 100*j))
+    sys.stdout.flush()  
```

Comment:
perhaps this belongs in a more generic utilities file, as we'll want to use progress bars for other things, and the previous way @bethem implemented it was not compatible with dask

Reply:
(Note from discussion with Naomi and Nicole): this progress bar only works for for-loops, so we will leave it here for now